postreise/analyze/generation/capacity_value.py

```diff
@@ -57,3 +61,39 @@ def calculate_net_load_peak(scenario, resources, hours=100):
     net_demand = total_demand - resource_generation
     top_hours = net_demand.sort_values(ascending=False).head(hours).index
     return resource_generation[top_hours].mean()
+
+
+def get_capacity_by_resources(scenario, area, resources, area_type=None):
+    """Get total capacity value of certain resources in the specific area of a
+    scenario.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of: *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources
+    :param str area_type: one of: *'loadzone'*, *'state'*,
+        *'state_abbr'*, *'interconnect'*
+    :return: (*pandas.Series*) -- index: resources, column: total capacity values
+    """
+    plant_id = get_plant_id_for_resources_in_area(
+        scenario, area, resources, area_type=area_type
+    )
+    grid = scenario.state.get_grid()
+
+    return grid.plant.loc[plant_id].groupby("type")["Pmax"].sum()
+
+
+def get_storage_capacity(scenario, area, area_type=None):
+    """Get total storage capacity value in the specific area of a scenario.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of: *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str area_type: one of: *'loadzone'*, *'state'*,
+        *'state_abbr'*, *'interconnect'*
+    :return: (*float*) -- total storage capacity value
+    """
+    grid = scenario.state.get_grid()
+    storage_id = get_storage_id_in_area(scenario, area, area_type)
+
+    return sum(grid.storage["gen"].loc[storage_id].Pmax.values)
```

Comment:
is this equivalent to `grid.storage["gen"].loc[storage_id].Pmax.sum()`?

Reply:
I believe so.

Reply:
Done.